### ISSUE TITLE
Incorrect error for explicitly defined number values

### ISSUE THREAD

**Issue by ChrisMBarr** [darraghoriordan/eslint-plugin-nestjs-typed#187]
I have a DTO property that allows a number to be 1-5 values only.  I am validating this with the class validator and providing explicit TS typing for when I use it in code

```ts
@IsNumber()
@Min(1)
@Max(5)
severity!: 1 | 2 | 3 | 4 | 5;
```

I feel like I am validating this correctly, but I am getting this linting error

> A non-primitive property with validation should probably use a @Type decorator. If this is an enum use @IsEnum(). If this is a plain object use @IsObject().eslint[@darraghor/nestjs-typed/validated-non-primitive-property-needs-type-decorator](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed/blob/main/src/docs/rules/validated-non-primitive-property-needs-type-decorator.md)

My guess is that this rule is only expecting `number` as the type and doesn't know what to do with the explicit number values `1 | 2 | 3 | 4 | 5` specified here

**Comment by darraghoriordan** [comment]
Yea it should be straightforward to update the rule to look for "BinaryExpression" with only primitives in the AST to handle these. Feel free to submit a PR if you get a chance. 

Alternatively you could create a type as the rule suggests
```
export type MyOneToFiveType =  1 | 2 | 3 | 4 |  5;
```

and then `severity!: MyOneToFiveType` shouldn't complain anymore

**Comment by ChrisMBarr** [comment]
PR #188 has resolved this issue for me!
